src/Components/Health/Checker/PerformanceChecker/FineGrainedCachingChecker.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Frosh\Tools\Components\Health\Checker\PerformanceChecker;
+
+use Frosh\Tools\Components\Health\Checker\CheckerInterface;
+use Frosh\Tools\Components\Health\HealthCollection;
+use Frosh\Tools\Components\Health\SettingsResult;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+class FineGrainedCachingChecker implements PerformanceCheckerInterface, CheckerInterface
+{
+    public const DOCUMENTATION_URL = 'https://developer.shopware.com/docs/guides/hosting/performance/performance-tweaks.html#disable-fine-grained-caching';
+
+    public function __construct(
+        #[Autowire('%kernel.shopware_version%')]
+        public readonly string $shopwareVersion,
+        #[Autowire('%shopware.cache.tagging.each_config%')]
+        public readonly bool $cacheTaggingEachConfig,
+        #[Autowire('%shopware.cache.tagging.each_snippet%')]
+        public readonly bool $cacheTaggingEachSnippet,
+        #[Autowire('%shopware.cache.tagging.each_theme_config%')]
+        public readonly bool $cacheTaggingEachThemeConfig,
+    ) {}
+
+    public function collect(HealthCollection $collection): void
+    {
+        if (\version_compare('6.5.4.0', $this->shopwareVersion, '>')) {
```

Comment:
I would rather inject all parameters and then check it with hasParameter. When you are below this version these parameters are not existing and the container cannot be built anymore

Reply:
I have fixed it like in https://github.com/FriendsOfShopware/FroshTools/pull/277, we had the same problem there.